crypto/s2n_hmac.c

```diff
@@ -268,6 +268,7 @@ int s2n_hmac_update(struct s2n_hmac_state *state, const void *in, uint32_t size)
     const uint32_t HIGHEST_32_BIT = 4294949760;
     ENSURE_POSIX(size <= (UINT32_MAX - HIGHEST_32_BIT), S2N_ERR_INTEGER_OVERFLOW);
     uint32_t value = (HIGHEST_32_BIT + size) % state->hash_block_size;
+    ENSURE_POSIX(value <= (UINT32_MAX - state->currently_in_hash_block), S2N_ERR_INTEGER_OVERFLOW);
```

Comment:
Should we be using [`s2n_add_overflow`](https://github.com/awslabs/s2n/blob/main/utils/s2n_safety.c#L221) instead?

Reply:
We could, let me give that a try! @camshaft

Reply:
Good catch! Implemented as suggested. @camshaft